### PR TITLE
Use npm shrinkwrap.

### PR DESCRIPTION
**Pull request by idralyuk**
Addresses https://github.com/ql-io/ql.io/issues/303

### CODE REVIEW

**Comment by s3u** [comment]
Could you confirm if you tested new apps (outside our source tree)?

File: modules/engine/package.json [hunk 1]
```diff
@@ -11,25 +11,25 @@
     },
     "main": "lib/engine.js",
     "dependencies": {
-        "winston": "0.5.9",
+        "winston": "0.5.11",
         "underscore": "1.3.1",
-        "xml2json": "https://github.com/ql-io/node-xml2json/tarball/master",
+        "xml2json": "0.2.4",
```
Comment (s3u) [inline modules/engine/package.json:16]: My pull req for coercion is still open - see https://github.com/buglabs/node-xml2json/pull/16. xml2json@0.2.4 breaks type coercion support needed. Can you clarify why this needs to be changed from the tarball to @0.2.4?

Comment (s3u) [inline modules/engine/package.json:16]: It is likely that there are no special tests to check type coercion in engine. We need to add one.

Secondly, could you explain if shrinkwrap is unable to process tarball references?

Comment (s3u) [inline modules/engine/package.json:16]: I see the issue. package.json and npm-shrinkwrap.json are not using the same version of xml2json. So, this pull req gets the right xml2json if I use npm 1.1.12, but the wrong one if I use an older version of npm. This needs to be fixed.

Could you also broadcast an email to the team that we need to redo shrinkwrap whenever we change one of our modules - for instance, engine needs to updated when compiler changes.

**Comment by idralyuk** [comment]
Just sent the email to the team.
Now testing the apps.
Correct: looks like shrinkwrap is unable to process tarball references.

**Comment by s3u** [comment]
When I tested, this change fails to work on node installations that bundle pre-shrinkwrap versions. Reverting the package.json change to use the xml2json tarbal ref should fix.

**Pull request merged**